wandb/sdk/internal/system/assets/gpu_amd.py

```diff
@@ -0,0 +1,190 @@
+        try:
+            raw_stats = get_rocm_smi_stats()
+            cards = []
+
+            for key in sorted(raw_stats.keys()):
+                if not key.startswith("card"):
+                    continue
+                card_stats = raw_stats[key]
+
+                stats: _Stats = {
+                    "gpu": float(card_stats["GPU use (%)"]),
+                    "memoryAllocated": float(card_stats["GPU memory use (%)"]),
+                    "temp": float(card_stats["Temperature (Sensor memory) (C)"]),
+                    "powerWatts": float(
+                        card_stats["Average Graphics Package Power (W)"]
+                    ),
+                    "powerPercent": (
+                        float(card_stats["Average Graphics Package Power (W)"])
+                        / float(card_stats["Max Graphics Package Power (W)"])
+                        * 100
+                    ),
+                }
+
+                cards.append(stats)
+
+            self.samples.append(cards)
+
+        except (OSError, ValueError, TypeError, subprocess.CalledProcessError) as e:
+            logger.exception(f"GPU stats error: {e}")
+
+    def clear(self) -> None:
+        self.samples.clear()
+
+    def aggregate(self) -> dict:
+        if not self.samples:
+            return {}
+        stats = {}
+        device_count = len(self.samples[0])
+
+        for i in range(device_count):
+            samples = [sample[i] for sample in self.samples]
+
+            for key in samples[0].keys():
+                samples_key = [s[key] for s in samples]  # type: ignore
+                aggregate = aggregate_mean(samples_key)
+                stats[self.name.format(gpu_id=i, key=key)] = aggregate
+
+        return stats
+
+
+@asset_registry.register
+class GPUAMD:
+    """GPUAMD is a class for monitoring AMD GPU devices.
+
+    Uses AMD's rocm_smi tool to get GPU stats.
+    For the list of supported environments and devices, see
+    https://github.com/RadeonOpenCompute/ROCm/blob/develop/docs/deploy/
+    """
+
+    def __init__(
+        self,
+        interface: "Interface",
+        settings: "SettingsStatic",
+        shutdown_event: threading.Event,
+    ) -> None:
+        self.name = self.__class__.__name__.lower()
+        self.metrics: List[Metric] = [
+            GPUAMDStats(),
+        ]
+        self.metrics_monitor = MetricsMonitor(
+            self.name,
+            self.metrics,
+            interface,
+            settings,
+            shutdown_event,
+        )
+        telemetry_record = telemetry.TelemetryRecord()
+        telemetry_record.env.amd_gpu = True
+        interface._publish_telemetry(telemetry_record)
+
+    @classmethod
+    def is_available(cls) -> bool:
+        try:
+            rocm_smi_available = shutil.which(ROCM_SMI_CMD) is not None
+            if rocm_smi_available:
+                _ = get_rocm_smi_stats()
+                return True
+        except Exception:
+            pass
+        return False
```

Comment:
nit: This seems indirect.  What part here might raise an exception (seems like it would get `get_rocm_smi_stats()`?

Reply:
fixed